packages/racket/rktio_network.c.patch

```diff
@@ -0,0 +1,28 @@
+From 78fbcab276ffdfd8525a00e320dff81ffcbae4ce Mon Sep 17 00:00:00 2001
+From: Matthew Flatt <mflatt@racket-lang.org>
+Date: Fri, 8 Nov 2024 10:29:00 -0700
+Subject: [PATCH] rktio: avoid `AI_V4MAPPED` on Android
+
+Although `AI_V4MAPPED` is `#define`d, using it trigger a "Invalid
+value for ai_flags" error.
+---
+ racket/src/rktio/rktio_network.c | 4 ++--
+ 1 file changed, 2 insertions(+), 2 deletions(-)
+
+diff --git a/racket/src/rktio/rktio_network.c b/racket/src/rktio/rktio_network.c
+index 2a1d6731fa3..c73fc32977f 100644
+--- a/src/rktio/rktio_network.c
++++ b/src/rktio/rktio_network.c
+@@ -727,10 +727,10 @@ rktio_addrinfo_lookup_t *rktio_start_addrinfo_lookup(rktio_t *rktio,
+   /* On modern systems, `AI_ADDRCONFIG` and `AI_V4MAPPED` tend to be
+      the defaults when `hints` is NULL, so add them if they seem
+      available. */
+-#ifdef AI_ADDRCONFIG
++#if defined(AI_ADDRCONFIG)
```

Comment:
Isn't `#ifdef` equivalent to `#if defined()`,
why do we patch this here as well if this is not checking for Android?

Reply:
This is an upstream patch. Maybe it makes them happy...

Reply:
Oh I didn't realize it came from upstream.
Best to just use it as is in that case.